include/bmi/AbstractCLibBmiAdapter.hpp

```diff
@@ -192,24 +97,18 @@ namespace models {
              * @throws ``::external::ExternalIntegrationException`` If symbol could not be found for the shared library.
              * @see dynamic_load_symbol(std::string, bool)
              */
-            inline void *dynamic_load_symbol(const std::string &symbol_name) {
-                return dynamic_load_symbol(symbol_name, false);
-            }
+            void *dynamic_load_symbol(const std::string &symbol_name);
 
-            inline const std::string &get_bmi_registration_function() {
-                return bmi_registration_function;
-            }
+            const std::string &get_bmi_registration_function();
 
-            inline const void *get_dyn_lib_handle() {
-                return dyn_lib_handle;
-            }
```

Comment:
I could've replaced `inline` with `__attribute__((always_inline))` here instead to get the intended behavior, but opted to move definitions to a source file instead because it probably isn't worth it currently.

Reply:
I would consider keeping at least these three small, simple functions in the header and marked `inline`.  They are relatively simple wrapper functions which are used for each adapter and at scale (800k calls at initialization of conus) could save some overhead in intialization.  The library load and load symbol functions are more complex, and inlining them would likely have some similar initializaiton optimizations avoiding additional function calls in setting up the adapters.  I don't think these are called from many different places, so the resulting binary size and build times should be minimal.
We may want to at least profile some large init times and see what the overhead in intialization is from NOT inlining these.

Reply:
Reverted in [7fc98b0](https://github.com/NOAA-OWP/ngen/pull/750/commits/7fc98b0dd9aa81f6da5a40c160a9d619ad9a3038)